python/hooks/check.bash

```diff
@@ -35,7 +35,7 @@ standardTests() (
 
     echo -e "\n\x1b[1;36mPylint:\x1b[0m"
     # shellcheck disable=SC2046,SC2086
-    pylint ${pylintArgs:-} --recursive=y . 2>&1 | sed 's/^/  /'
+    HOME=$TMP pylint ${pylintArgs:-} --recursive=y . 2>&1 | sed 's/^/  /'
```

Comment:
For the sake of determinism, maybe we should create a tempdir and nuke it after pylint has run. Otherwise, we run the risk of actually getting cached data between runs which should work but...

Reply:
For re-runs in the same shell it would use the cache yes. But isn't that kind of the point of a cache and should be expected to work? For build (and each new shell) it will be a unique fresh directory every time.

Reply:
But $TMP is not changed for each build?

Reply:
Yes, nix sets it to the build directory, from https://nixos.org/manual/nix/stable/language/derivations.html:

> A temporary directory is created under the directory specified by TMPDIR (default /tmp) where the build will take place. The current directory is changed to this directory.
> ...
> * Also, TMPDIR, TEMPDIR, TMP, TEMP are set to point to the temporary directory. This is to prevent the builder from accidentally writing temporary files anywhere else. Doing so might cause interference by other processes.